### DISCUSSION TITLE
Slow site (latest version)

### DISCUSSION THREAD

**Discussion by wattaman** [FreshRSS/FreshRSS#5225] [Q&A]
Is it me or the latest version needs a little more resources to run properly?! It seems pages are loading much slower, sometimes ending in an error. Behind Cloudflare, this is 522 - this is because (CF says) it takes long time for the website to respond to Cloudflare requests.
I think I need to allow more resources, but not sure what. The version before update worked just fine with these settings (nginx + php7.4 + sqlite):
```
fastcgi_param   PHP_VALUE   "memory_limit = 64M;\n post_max_size = 10M;\n upload_max_filesize = 5M;\n max_execution_time = 30;\n pm.process_idle_timeout = 30s;";
fastcgi_cache_valid 25m;
fastcgi_cache_min_uses 1;
fastcgi_cache_lock on;
fastcgi_read_timeout 30s;
```

The FreshRSS installation page (_?c=update&a=checkInstall_) doesn't have any warning, everything is green there. 

A speed test in gtmetrix.com shows that this URI is taking 19 seconds when loading the category/feed page _?a=normal&get=f_215&rid=641dd57dabd34&next=1679553018622352&state=3&ajax=1_ 

Any suggestions?

**Comment by math-GH** [comment]
Could you increase your PHP memory limit? Would it solve the issue?

**Comment by Alkarex** [comment]
19 seconds is much too long. Most likely, this is not related to PHP but to the underlying database. What database do you use?
The idea would be to run a couple of requests manually to see how long they take.

Example for PosgreSQL in Docker:

```console
$ docker exec postgres psql -d freshrss -U freshrss -A -c '\dt'
List of relations
Schema|Name|Type|Owner
public|alice_category|table|freshrss
public|alice_entry|table|freshrss
public|alice_entrytag|table|freshrss
public|alice_entrytmp|table|freshrss
public|alice_feed|table|freshrss
public|alice_tag|table|freshrss
public|demo_category|table|freshrss
...
(24 rows)

$ # Pick one of the `*_entry` tables you have (alice_entry in my example):

$ time docker exec postgres psql -d freshrss -U freshrss -A -c 'SELECT id, link, title FROM alice_entry WHERE is_read = 0 ORDER BY id DESC LIMIT 10;'
id|link|title
1679836043935486|https://techcrunch.com/2023/03/26/tech-company-layoffs-2023-morale/|The layoffs will continue until (investor) morale improves
...
(10 rows)

real    0m0.652s
user    0m0.021s
sys     0m0.039s
```
